### PR TITLE
Borrower service

### PR DESCRIPTION
**Pull request by wuaho**

### CODE REVIEW

File: app/services/borrower.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @user = user
+    @game = game
+  end
+
+  def user_can_borrow_game?
+    # Checks that:
+    #   1. User is not banned
+    #   2. User has member role or higher
+    #   3. User has no active borrowing
+    #   4. Game stock is bigger than number of active borrowings of that game
```
Comment (jonnyjava) [inline app/services/borrower.rb:14]: 🔥

File: app/services/borrower.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    user_not_banned? && user_is_member? && user_has_no_active_borrowing? && game_is_available?
+  end
+
```
Comment (jonnyjava) [inline app/services/borrower.rb:17]: private

File: spec/services/borrower_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      end
+    end
+    context 'when user already has an active borrowing' do
+      create(:borrowing)
+      it 'returns false' do
+        expect(subject.user_can_borrow_game?).to be false
+      end
+    end
+    context 'when game is not available' do
+      create(:borrowing)
```
Comment (jonnyjava) [inline spec/services/borrower_spec.rb:33]: `before { create(:borrowing) }`

**Review by jonnyjava — APPROVED** [review]

File: app/services/borrower.rb (hunk 1 above)
Comment (wuaho) [reply app/services/borrower.rb:14]: Fireeee

File: spec/services/borrower_spec.rb (hunk 3 above)
Comment (wuaho) [reply spec/services/borrower_spec.rb]: fixerino

File: app/services/borrower.rb (hunk 2 above)
Comment (wuaho) [reply app/services/borrower.rb:12]: privaterino

**Pull request merged**